crates/wasmparser/src/lib.rs

```diff
@@ -325,6 +325,7 @@ macro_rules! for_each_operator {
             @bulk_memory DataDrop { data_index: u32 } => visit_data_drop
             @bulk_memory MemoryCopy { dst_mem: u32, src_mem: u32 } => visit_memory_copy
             @bulk_memory MemoryFill { mem: u32 } => visit_memory_fill
+            @bulk_memory MemoryDiscard { mem: u32 } => visit_memory_discard
```

Comment:
The `@bulk_memory` here is used as a proposal to gate access to this instruction, so this'll want to be something other than `bulk_memory` since otherwise `memory.discard` will be enabled by default.

In lieu of an official proposal, could this perhaps be something like `@experimental` or `@memory_control`? That'll require adding a field to `WasmFeatures` which is ok and would be necessary for this anyway.

Reply:
I'll add `@memory_control` I think!

Reply:
Added a `@memory_control` feature. I hope I set true/false correctly in the various places where it was required.